ui/app/controllers/jobs/run/templates/new.js

```diff
@@ -0,0 +1,71 @@
+import { A } from '@ember/array';
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+const ALL_NAMESPACE_WILDCARD = '*';
+
+export default class JobsRunTemplatesController extends Controller {
+  @service router;
+  @tracked templateName = null;
+  @tracked templateNamespace = 'default';
+
+  keyValues = A([]);
+
+  get namespaceOptions() {
+    const namespaces = this.store
+      .peekAll('namespace')
+      .map(({ name }) => ({ key: name, label: name }));
+
+    if (namespaces.length <= 1) return null;
+
+    // Create default namespace selection
+    namespaces.unshift({
+      key: ALL_NAMESPACE_WILDCARD,
+      label: 'All (*)',
+    });
+
+    return namespaces;
+  }
+
+  @action
+  updateKeyValue(key, value) {
+    if (this.keyValues.find((kv) => kv.key === key)) {
+      this.keyValues.find((kv) => kv.key === key).value = value;
+    } else {
+      this.keyValues.pushObject({ key, value });
```

Comment:
I noticed that `Variables` have warnings. Are they needed here?

Reply:
They're helpful for storing information about problems in a variable (whether it comes back from a server, has a duplicate name, etc.) but are probably not needed here.